stream-parent/stream-core/src/main/java/org/interledger/stream/crypto/AesGcmStreamEncryptionService.java

```diff
@@ -0,0 +1,375 @@
+package org.interledger.stream.crypto;
+
+import org.interledger.core.SharedSecret;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import com.google.common.hash.Hashing;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.util.Arrays;
+import java.util.Objects;
+
+import javax.crypto.BadPaddingException;
+import javax.crypto.Cipher;
+import javax.crypto.IllegalBlockSizeException;
+import javax.crypto.NoSuchPaddingException;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.GCMParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+
+/**
+ * An {@link StreamEncryptionService} that uses a JavaKeystore for underlying key storage.
+ */
+public class AesGcmStreamEncryptionService implements StreamEncryptionService {
```

Comment:
This class is a copy/paste of the original JavaxStreamEncryptionService, with added support for encryption/decryption modes to support existing functionality in a fully backwards-compatible manner, plus standards-compliant functionality.